lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -828,6 +828,22 @@ namespace Js
                 isJitCandidate = true;
             }
         });
+
+        if (!isJitCandidate)
+        {
+            // Now check loop body entry points
+            this->MapLoopHeaders([&](uint loopNumber, LoopHeader* header)
+            {
+                header->MapEntryPoints([&](int index, LoopEntryPointInfo* entryPointInfo)
+                {
+                    if ((entryPointInfo->IsCodeGenPending() && isJitModeFunction) || entryPointInfo->IsCodeGenQueued() || entryPointInfo->IsCodeGenRecorded() || (entryPointInfo->IsCodeGenDone() && !entryPointInfo->nativeEntryPointProcessed))
+                    {
+                        isJitCandidate = true;
```

Comment:
>isJitCandidate = true; [](start = 24, length = 22)

break or return from the lambda, and no need to check the remaining loops if any?